src/modules/core/components/Popover/Popover.tsx

```diff
@@ -116,30 +120,35 @@ const Popover = ({
 
   const lastIsOpenProp = usePrevious(isOpenProp);
 
+  const close = useCallback(
+    (data?: any, modifiers?: { cancelled: boolean }) => {
+      if (window) {
+        window.clearTimeout(openTimeoutRef.current);
+        window.clearTimeout(closeTimeoutRef.current);
+      }
+      _setIsOpen(false);
+      if (typeof onClose == 'function') onClose(data, modifiers);
+    },
+    [onClose],
+  );
+
   const requestOpen = useCallback(() => {
     if (isOpen) {
       return;
     }
+    if (closeAfterDelay && window) {
+      closeTimeoutRef.current = window.setTimeout(() => {
+        close();
+      }, 3000 + (openDelay || 0));
```

Comment:
Is there any particular reason why you're using `window.setTimeout` instead of just `setTimeout` ?

Reply:
Great :+1:

Reply:
@rdig Ive done some investigation and I am not sure how I can avoid using Window when using setTimeout inside a useCallback. I saw that in StepMetaMask.tsx that the same technique is used where the timeout is assigned to a variable with the type useRef<number>(). Is there someway to not use window inside a useCallback or should I restructure this somehow to do the setTimeout outside of the useCallback?

One option is to change the variable, const openTimeoutRef = useRef<any>(null); and that will allow using setTimeout without window but I am not sure using any is the best solution or is that ok for this case?

Reply:
@Julianahlk You can use the Node `setTimeout` and cast the return value as a `number` for use with `clearTimeout`. These appeases the ts compiler and means you can dispense with the references to `window`. 

See: https://stackoverflow.com/questions/45802988/typescript-use-correct-version-of-settimeout-node-vs-window

I believe this would resolve your problem?

Reply:
Thanks Will! It did solve the problem.